src/Image_Backend.php

```diff
@@ -84,87 +78,66 @@ public function setImageResource($resource);
      * @return array Tuple associative array (Filename, Hash, Variant) Unless storing a variant, the hash
      * will be calculated from the given data.
      */
-    public function writeToStore(AssetStore $assetStore, $filename, $hash = null, $variant = null, $config = []);
+    public function writeToStore(AssetStore $assetStore, string $filename, ?string $hash = null, ?string $variant = null, array $config = []): array;
 
     /**
      * Write the backend to a local path
      *
      * @param string $path
      * @return bool if the write was successful
      */
-    public function writeTo($path);
+    public function writeTo(string $path): bool;
 
     /**
      * Set the quality to a value between 0 and 100
-     *
-     * @param int $quality
      */
-    public function setQuality($quality);
+    public function setQuality(int $quality): static;
+
+    /**
+     * Get the current quality (between 0 and 100)
+     */
+    public function getQuality(): int;
```

Comment:
This method is in `InterventionBackend` and we use it in tests without checking if the object is `InterventionBackend` or some other class using this interface. Seems sensible to just add it as a required method.